2019-10-26  Kimberly Lin Shi  <kimberls@lnxsrv07.seas.ucla.edu>

	* timespec.h (timespec_cmp): Modified timespec_cmp function so that there would not be any overflow/unwanted behavior when doing subtraction on the two dates.


--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2019-10-26 16:20:40.687737000 -0700
@@ -46,7 +46,8 @@
 timespec_cmp (struct timespec a, struct timespec b)
 {
   int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  return (a.tv_sec > b.tv_sec) ? 1: (a.tv_sec < b.tv_sec) ? -1 : diff;
+  //return diff ? diff : a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
